configure.ac

```diff
@@ -1635,26 +1631,12 @@ if test -n "$perform_dbgmutex" ; then
 fi
 
 pac_cross_compiling=no
+# POSIX shell will not source file unless it is in the PATH
+PATH=$PATH:.
 if test "$with_cross" != "no" ; then
     if test -s "$with_cross" ; then
         AC_MSG_RESULT([Reading values from cross-compilation file $with_cross])
         . $with_cross
-	# Autoconf 2.52 no longer sets cross_compiling except with the
-	# awkward "targethost" options.
-	pac_cross_compiling=yes
-	cross_compiling=yes
-	ac_cv_prog_cc_cross=yes
-	ac_cv_prog_f77_cross=yes
- 	ac_cv_prog_fc_cross=yes
-        ac_cv_prog_cxx_cross=yes
-        export cross_compiling
-	# Export all cross variables.  Any subsidiary configure should also
-        # export CROSS_xxx
-	rm -f confcross
-	(set) 2>&1 | grep CROSS_ | \
-	      sed -e 's/^/export /g' -e 's/=.*//g' > confcross
-	. confcross
-	rm -f confcross
     fi
```

Comment:
To coordinate between sub-configure is just too messy. I suggest a solution is to have a clean output from MPICH's configure --`mpichconf.h` or even `mpich_config.sh`  -- that other modules can parse or source. That way, we'll have a consolidated place to coordinate between packages.

For datatypes, currently we set unavailable datatypes to `MPI_DATATYPE_NULL`. Can we instead just expose `HAVE_type` macros in `mpi.h`? Either is non-standard interface, but if we are going to use it -- dtpool, romio -- we might as well use a straight-forward mechanism. We might as well expose the datatype sizes and c type conversions too.

Reply:
`HAVE_type` would not be portable between different MPI libraries.